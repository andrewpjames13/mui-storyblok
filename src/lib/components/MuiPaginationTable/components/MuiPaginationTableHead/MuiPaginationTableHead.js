import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { TableHead } from '@material-ui/core';
import MuiPaginationTableRow from '../MuiPaginationTableRow/MuiPaginationTableRow';
import StoryBlok from '../../../../utils/Storyblok';

const MuiPaginationTableHead = ({
  rootClass,
  content,
}) => {
  const components = {
    MuiPaginationTableRow,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  return (
    <TableHead className={styles.root}>
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </TableHead>
  );
};

export default MuiPaginationTableHead;

MuiPaginationTableHead.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
   * Content passed to render
   * components:
    MuiPaginationTableRow,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiPaginationTableHead.defaultProps = {
  rootClass: [],
};
