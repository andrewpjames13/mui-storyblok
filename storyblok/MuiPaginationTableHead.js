module.exports = {
  "name": "MuiPaginationTableHead",
  "display_name": null,
  "updated_at": "2020-04-21T21:30:50.127Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiPaginationTableRow"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiPaginationTableHead",
  "component_group_uuid": null
}