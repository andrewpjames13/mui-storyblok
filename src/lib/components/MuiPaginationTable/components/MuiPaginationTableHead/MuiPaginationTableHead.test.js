import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiPaginationTableHead from './MuiPaginationTableHead';

function setup() {
  const props = {
    content: [
      {
        component: 'MuiPaginationTableRow',
        content: [
          {
            component: 'MuiTableCell',
            content: [
              {
                component: 'MuiTypography',
                content: [
                  {
                    component: 'MuiText',
                    text: 'wooo',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  };

  const comp = shallow(<MuiPaginationTableHead {...props} />);
  return { comp, props };
}

describe('<MuiPaginationTableHead />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiPaginationTableHead {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render MuiPaginationTableHead', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
