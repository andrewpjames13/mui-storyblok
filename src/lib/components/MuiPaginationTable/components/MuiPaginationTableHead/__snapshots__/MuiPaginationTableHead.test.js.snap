// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTableHead /> snapshot 1`] = `
<thead
  className="MuiTableHead-root makeStyles-root-2"
>
  <tr
    className="MuiTableRow-root makeStyles-root-4 MuiTableRow-head"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head makeStyles-root-10"
      scope="col"
    >
      <p
        className="MuiTypography-root makeStyles-root-23 MuiTypography-body1"
      >
        wooo
      </p>
    </th>
  </tr>
</thead>
`;
